### DISCUSSION TITLE
pinia provideApolloClient() add

### DISCUSSION THREAD

**Discussion by productdevbook** [vuejs/pinia#819] [Help and Questions] [answered]
<img width="1038" alt="CleanShot 2021-11-22 at 17 55 32@2x" src="https://user-images.githubusercontent.com/38668796/142883177-6d2273f8-e73c-46d7-b393-0b57106fa771.png">

Can a document be written or a plugin made for it? Apollo - graphql

**Comment by productdevbook** [comment]
```
import {provideApolloClient} from '@vue/apollo-composable'
import apolloClient from '@/composables/apollo'


 actions: {
    async hydrate(data: LoginMutationMutation) {
      provideApolloClient(apolloClient)
    }
```

`   provideApolloClient(apolloClient)`I'm adding it one by one in his action. There has to be an easier way.

**Reply by vsqla** [reply]
import apolloClient from '@/composables/apollo'
and what inside this?

**Comment by posva** [comment, marked as answer]
Use any copmosition function inside of a setup store:

```js
defineStore('id', () => {
  useApollo()
  provideApollo(...)

  return {
    // anything you want to expose like a component
  }
})
```
